### ISSUE TITLE
Use IRC syntax table

### ISSUE THREAD

**Issue by jorgenschaefer** [jorgenschaefer/circe#181]
IRC has its own idea of what constitutes a nick, which is different from a word in natural language. Circe should help extension authors match full nicks by defining all valid characters in a nick as symbol constituents in the syntax table. This means `"\\s_+"` will match valid nicks.

**Comment by wasamasa** [comment]
Is that intended to clean up code like https://github.com/jorgenschaefer/circe/commit/ae4ea7e68a3a5fdc0073496a5c5334203f2d14e6?

**Comment by jorgenschaefer** [comment]
Yup. Basically, use symbol syntax instead of word syntax, and be able to use it everywhere, always, even for `C-M-f` and such.
